content/docs/how-we-work/drupal/config.md

```diff
@@ -14,3 +14,7 @@ The `production` should not have `views_ui` and `field_ui` module installed. The
 ## Ignoring configurations
 
 There is a configuration which you want to ignore whenever you are exporting configurations from your setup. Use [`config_ignore`](https://www.drupal.org/project/config_ignore).
+
+## Managing Secret Keys
+
+We use modules like [Key](https://www.drupal.org/project/key) to manage secrets keys (such as API and encryption keys).
```

Comment:
Nice, but the thing is that we don't as far as I know. Do you know of any of our projects using key?

As per my knowledge, the sites we host on platform.sh have a different mechanism for storing keys (still secure). There are a variety of other mechanisms for other hosts but I don't know of any project using key. I am hoping they would but we should first find out.

Reply:
> Nice, but the thing is that we don't as far as I know. Do you know of any of our projects using key?
> 
> As per my knowledge, the sites we host on platform.sh have a different mechanism for storing keys (still secure). There are a variety of other mechanisms for other hosts but I don't know of any project using key. I am hoping they would but we should first find out.

@axelabhay: Have your say on this?

Reply:
We received this suggestion while collaborating contents for `Security` via this [document](https://docs.google.com/document/d/1eA7IVZwF3kokfa8uEkNudHu6VRQrfr4-cgC59YEmHBA/edit?disco=AAAA5ZFNqSk).

I am not sure if we use this module frequently in our projects. I am okay to close this PR and we may consider this content while we talk more about different mechanism of storing keys in EHB.

Cc: @Kalaiselvan88

Reply:
> So, why not bring in other options as well? 
Agree. To do.

Reply:
@hussainweb, I am not sure of what `secure stores` are all about; when you are saying 'It works with secure stores and that's not mentioned here'.

Also, I tried following paragraphs for `Managing Secret Keys` section contents, please share your thoughts.
Cc: @Kalaiselvan88 @Yemaneberhan-Lemma 

> Storing secret/API keys in the codebase(configuration files) or databases may expose them to unauthorised access, potentially leading to data breaches or misuse of the API services.
> 
> We connect to the production server via ssh and manage to store secrets as external files, which are not part of the codebase or database and can be easily managed and updated. This may include following certain conventions regarding the location and naming of these files on different hosting platforms. Secrets can also be stored in environment variables, but storing as external file method provides a higher level of security than environment variables as any code can inspect environment variables.
> 
> Drupal also provides [Key](https://www.drupal.org/project/key) module for managing sensitive keys, which allows site administrators to define how and where keys are stored, providing a high level of security and meeting regulatory or compliance requirements.

Reply:
> Secrets can also be stored in environment variables, but storing as external file method provides a higher level of security than environment variables as any code can inspect environment variables.

I wonder where you found this. This is not true in my experience. If the environment variables are set correctly, it has similar security to what file access might bring.

Reply:
That said, it's a good improvement. Also, sorry about "secure stores". I didn't know what else to call them. I am essentially referring to services like Lockr (a product by the folks who wrote the key module), 1Password, Ansible Vault, etc. They are designed to store sensitive data securely and even provide ways to transmit them securely until it reaches the application's boundary.

Reply:
> > Secrets can also be stored in environment variables, but storing as external file method provides a higher level of security than environment variables as any code can inspect environment variables.
> 
> I wonder where you found this. This is not true in my experience. If the environment variables are set correctly, it has similar security to what file access might bring.

This particular [point](https://www.cloudtruth.com/blog/the-pitfalls-of-using-environment-variables-for-config-and-secrets#:~:text=Environment%20variables%20are%20globally%20accessible%20to%20all%20processes%20running%20on%20the%20same%20machine.%20This%20means%20that%20if%20an%20attacker%20manages%20to%20infiltrate%20one%20application%2C%20they%20can%20potentially%20gain%20access%20to%20sensitive%20information%20stored%20in%20the%20environment%20variables%20of%20other%20applications.) supports it. 

Please check, and suggest what can be adjusted here. I have not tested it extensively but my thought says when compared to ENV variables learning about the secret file could be one level deeper exercise for attackers. :)

Reply:
That's true. But obscurity is just security theatre and, for a deliberate attack, both are the same. I can see that storing secrets in files might be safer against developer mistakes (like logging the entire env) but not more secure against an attacker. The only place files help is a fly-by attacker who is only looking at common places (like env, user profile files, default passwords, etc). And at that point, they can do almost anything.

For added perspective, see https://circleci.com/blog/jan-4-2023-incident-report/. This discussion is OT, however.

For this text, I suggest not calling files more secure. The phrase "any code can inspect environment variables" implies that the same code cannot read files, which is, of course, not true. I would say that dedicated secure stores are the safest (still breakable) followed by storing secrets directly in files or environment variables. All of these have caveats but that may be too detailed for our purposes.

Reply:
These points are helpful to understand the topic better, thank you @hussainweb. 

I think, below content is ready to go, @Yemaneberhan-Lemma.

> Storing secret/API keys in the codebase(configuration files) or databases may expose them to unauthorised access, potentially leading to data breaches or misuse of the API services.
> 
> We connect to the production server via ssh and manage to store secrets in environment variables or in files, which are not part of the codebase or database and can be easily managed and updated. Storing secrets in files may include following certain conventions regarding the location and naming of these files on different hosting platforms.
> 
> Drupal also provides [Key](https://www.drupal.org/project/key) module for managing sensitive keys, which allows site administrators to define how and where keys are stored, providing a high level of security and meeting regulatory or compliance requirements.